api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -229,125 +174,49 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
+
+    bigQueryJobConfig.forEach((domain, queryJobConfiguration) -> {
+      queryJobConfiguration.getNamedParameters().forEach((key, value) ->
+              parameters.add(generateResponseFromQueryParameter(key, value)));
+      respQueryList.add(new DataSetQuery()
+          .domain(Domain.fromValue(domain))
+          .query(queryJobConfiguration.getQuery())
+          .namedParameters(parameters));
+    });
     return ResponseEntity.ok(new DataSetQueryList().queryList(respQueryList));
   }
 
-  @VisibleForTesting
-  ValuesLinkingPair getValueSelectsAndJoins(List<DomainValuePair> valueSetList, Domain d) {
-    List<String> values = valueSetList.stream().map(valueSet -> valueSet.getValue())
-        .collect(Collectors.toList());
-    values.add(0, "CORE_TABLE_FOR_DOMAIN");
-    String domainAsName = d.toString().charAt(0) + d.toString().substring(1).toLowerCase();
-
-    String valuesQuery = "SELECT * FROM `${projectId}.${dataSetId}.ds_linking` WHERE DOMAIN = @pDomain AND DENORMALIZED_NAME in unnest(@pValuesList)";
-    Map<String, QueryParameterValue> valuesQueryParams = new HashMap<>();
-
-    valuesQueryParams.put("pDomain", QueryParameterValue.string(domainAsName));
-    valuesQueryParams.put("pValuesList", QueryParameterValue.array(values.toArray(new String[0]), String.class));
-
-    TableResult valuesLinking = bigQueryService.executeQuery(
-        bigQueryService
-            .filterBigQueryConfig(QueryJobConfiguration
-                .newBuilder(valuesQuery)
-                .setNamedParameters(valuesQueryParams)
-                .setUseLegacySql(false)
-                .build()));
-
-    List<String> valueJoins = new ArrayList<>();
-    List<String> valueSelects = new ArrayList<>();
-    valuesLinking.getValues().forEach((value) -> {
-      valueJoins.add(value.get("JOIN_VALUE").getStringValue());
-      if (!value.get("OMOP_SQL").getStringValue().equals("CORE_TABLE_FOR_DOMAIN")) {
-        valueSelects.add(value.get("OMOP_SQL").getStringValue());
-      }
+  @Override
+  public ResponseEntity<DataSetPreviewResponse> previewQuery(String workspaceNamespace,
+      String workspaceId, DataSetRequest dataSet) {
+    workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(workspaceNamespace, workspaceId, WorkspaceAccessLevel.READER);
+    DataSetPreviewResponse previewQueryResponse = new DataSetPreviewResponse();
+    Map<String, QueryJobConfiguration> bigQueryJobConfig = dataSetService.generateQuery(dataSet);
+    bigQueryJobConfig.forEach((domain, queryJobConfiguration) -> {
+      String query = queryJobConfiguration.getQuery().concat(" LIMIT "+ NO_OF_PREIVEW_ROWS);
+      queryJobConfiguration = queryJobConfiguration.toBuilder().setQuery(query).build();
+
+      TableResult queryResponse = bigQueryService.executeQuery(bigQueryService
+          .filterBigQueryConfig(queryJobConfiguration));
+
+      // Construct a list of all values with empty results.
+      List<DataSetPreviewValueList> valuePreviewList =
```

Comment:
I had to change the logic around here because it wasn't working with multiple domains. @NehaBroad please review to make sure it looks good to you!